crates/wit-encoder/Cargo.toml

```diff
@@ -10,9 +10,14 @@ version.workspace = true
 [lints]
 workspace = true
 
+[features]
+default = ["serde"]
+serde = ["dep:serde", "semver/serde"]
```

Comment:
I talked with @MendyBerger about this change and this PR and got more context on the rationale. I now think this is reasonable to land in wasm-tools, but I wanted to highlight a change here that @MendyBerger and I also talked about.

Specifically I think that this is a good location to document what this feature is and what it should be used for. Basically explaining that this enables, for example, json serialization/deserialization of `wit-encoder` structures. I think it would be worthwhile to comment here as well that this json format is different from the format of `wit-parser` and additionally serves a different purpose. For example if you're looking to create WIT things from json then this is the crate and feature for you, but if you're parsing WIT and reading the output through json then `wit-parser` is probably the better option.

Another thing I think would be good to mention here as well is that the exact structure of the JSON is probably not going to be super stable over time, so slightly tweaks and variants should be expected as this crate evolves.

Reply:
@alexcrichton I've added the notes you requested